Rubberduck.CodeAnalysis/Inspections/Concrete/EnumerationDeclaredWithinWorksheetInspection.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Rubberduck.CodeAnalysis.Inspections.Abstract;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.Resources.Inspections;
+using Rubberduck.VBEditor.ComManagement;
+using Rubberduck.VBEditor.SafeComWrappers;
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete
+{
+    internal class EnumerationDeclaredWithinWorksheetInspection : DeclarationInspectionBase
+    {
+        private readonly IProjectsProvider _projectsProvider;
+
+        public EnumerationDeclaredWithinWorksheetInspection(IDeclarationFinderProvider declarationFinderProvider, IProjectsProvider projectsProvider)
+            : base(declarationFinderProvider)
+        {
+            _projectsProvider = projectsProvider;
+        }
+
+        protected override bool IsResultDeclaration(Declaration declaration, DeclarationFinder finder)
+        {
+            return declaration.DeclarationType == DeclarationType.Enumeration
+                && declaration.QualifiedModuleName.ComponentType == ComponentType.Document;
+        }
```

Comment:
`IsResultDeclaration` should flag `Public` `Enumerations` only.  Duplicated `Private` enums (resulting from copying  a worksheet) will not result in compile error.